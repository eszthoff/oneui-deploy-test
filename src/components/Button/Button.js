import React from 'react';
import PropTypes from 'prop-types';
import bem from '../../packages/bem';
import styles from './Button.scss';
import { CONTEXTS, SIZES } from '../../constants';

const Button = props => {
    const { children, disabled, htmlAttributes, onClick, style, type, block } = props;
    return (
        <button
            {...block()}
            type={type}
            style={style}
            onClick={onClick}
            disabled={disabled}
            {...htmlAttributes}
        >
            {children}
        </button>
    );
};

Button.propTypes = {
    /** The label of the button */
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
    /** The button context (e.g. brand, primary, bad, good etc. - defaults to brand) */
    context: PropTypes.oneOf([...CONTEXTS, 'link']),
    /** The size of the button */
    size: PropTypes.oneOf(SIZES),
    /** whether or not to show block - level button(full width) */
    isBlock: PropTypes.bool,
    /** should button be disabled or not */
    disabled: PropTypes.bool,
    /** type of the button */
    type: PropTypes.oneOf(['submit', 'button']),
    /** optional html attrinutes such as target, href, mouseOver etc. */
    htmlAttributes: PropTypes.objectOf(
        PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.bool])
    ),
    /** callback function on click */
    onClick: PropTypes.func,
    /** inline styles */
    style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
    /** additional class names */
    className: PropTypes.string
};

Button.defaultProps = {
    context: 'neutral',
    size: 'normal',
    isBlock: false,
    disabled: false,
    type: 'button',
    htmlAttributes: {},
    onClick: null,
    style: null,
    className: ''
};

Button.propsToMods = ['context', 'size', 'isBlock'];

export default bem(styles)(Button);
