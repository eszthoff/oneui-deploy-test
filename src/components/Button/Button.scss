@import '../../themes/nice/nice.scss';

.button {
    padding: $spacing-medium $spacing-large;
    border-radius: $border-radius;
    text-align: center;
    color: $color-background;

    &--size {
        &_small {
            font-size: $font-size-small;
            padding: $spacing-normal, $spacing-medium;
        }
        &_large {
            font-size: $font-size-small;
            padding: $spacing-large $spacing-xlarge;
        }
    }

    &--block {
        width: 100%;
    }

    &--context {
        &_neutral {
            background-color: $color-background;
            border: $color-neutral 1px solid;
            color: $color-foreground;

            &:hover {
                background-color: $color-neutral;
            }
        }

        &_brand {
            background-color: $color-brand;

            &:hover {
                background-color: $color-brand-75;
            }
        }

        &_primary {
            background-color: $color-primary;

            &:hover {
                background-color: $color-primary-75;
            }
        }

        &_accent {
            background-color: $color-accent;

            &:hover {
                background-color: $color-accent-75;
            }
        }

        &_info {
            background-color: $color-info;

            &:hover {
                background-color: $color-info-75;
            }
        }

        &_good {
            background-color: $color-good;

            &:hover {
                background-color: $color-good-75;
            }
        }

        &_warning {
            background-color: $color-warning;

            &:hover {
                background-color: $color-warning-75;
            }
        }

        &_bad {
            background-color: $color-bad;

            &:hover {
                background-color: $color-bad-75;
            }
        }
    }

    &:disabled,
    &[disabled] {
        opacity: 50%;
    }
}
