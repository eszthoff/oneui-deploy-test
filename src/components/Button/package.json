{
    "name": "button",
    "version": "0.0.0",
    "private": true,
    "main": "./button.js"
}