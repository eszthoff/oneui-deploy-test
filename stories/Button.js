import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { boolean, number, text, select, withKnobs } from '@storybook/addon-knobs';  // eslint-disable-line import/no-extraneous-dependencies
import Button from '../src/components/Button';
import { CONTEXTS, SIZES } from '../src/constants';

storiesOf('Button', module)
    .addDecorator(withKnobs)
    .add('button', () => (
        <Button
            context={select('Context', CONTEXTS, CONTEXTS[1])}
            size={select('Size', SIZES, SIZES[1])}
            block={boolean('Block', false)}
            type={select('Type', ['submit', 'button'], 'submit')}
            disabled={boolean('Disabled', false)}
        >
            Click me!
        </Button>
    ));
