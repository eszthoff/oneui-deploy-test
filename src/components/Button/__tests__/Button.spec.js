import React from 'react';
// import { mount } from 'enzyme';
import { toJson } from 'enzyme-to-json';
import Button from '../Button';

describe('<Button> that renders a button', () => {
    // console.log(Button)
    it('should render default button correctly', () => {
        // const wrapper = mount(<div>try</div>)
        expect(true).toBeTruthy();
        // const wrapper = mount(<Button>Click me</Button>)
        // expect(toJson(wrapper)).toMatchSnapshot();
    });
});
