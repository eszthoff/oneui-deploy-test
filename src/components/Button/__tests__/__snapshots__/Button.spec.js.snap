// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> that renders a button should render default button correctly 1`] = `
<Component
  block={false}
  className=""
  context="brand"
  disabled={false}
  htmlAttributes={Object {}}
  onClick={null}
  size="normal"
  style={Object {}}
  type="button"
>
  <Button
    block={false}
    className=""
    context="brand"
    disabled={false}
    htmlAttributes={Object {}}
    onClick={null}
    size="normal"
    style={Object {}}
    theme={Object {}}
    type="button"
  >
    <button
      className=""
      disabled={false}
      onClick={null}
      style={Object {}}
      type="button"
    >
      Click me
    </button>
  </Button>
</Component>
`;
